frb_codegen/src/utils/dart_repository/dart_repo.rs

```diff
@@ -83,28 +83,26 @@ impl DartRepository {
         package: &str,
         manager: DartDependencyMode,
         requirement: &VersionReq,
-    ) -> anyhow::Result<()> {
+    ) -> Result<(), crate::commands::Error> {
         let at = self.at.to_str().unwrap();
         debug!("Checking presence of {} in {} at {}", package, manager, at);
         let lock_file = read_file(at, DartToolchain::lock_filename())?;
         let lock_file: PubspecLock = serde_yaml::from_str(&lock_file).map_err(|e| {
-            anyhow::Error::msg(format!(
-                "unable to parse {} in {}: {:#}",
-                DartToolchain::lock_filename(),
-                at,
-                e
-            ))
+            anyhow!(
+                "unable to parse {} in {at}: {e:#}",
+                DartToolchain::lock_filename()
+            )
         })?;
         let dependency = lock_file.packages.get(package);
         let version = match dependency {
             Some(dependency) => {
                 let pm = dependency.installed_in();
                 if pm.as_ref() != Some(&manager) {
-                    return Err(anyhow::Error::new(Error::InvalidDep {
+                    return Err(Error::InvalidDep {
```

Comment:
nit: are we missing backtrace when migrating from anyhow error to thiserror?

seems we can add it https://stackoverflow.com/questions/61744018/how-do-i-print-a-backtrace-without-panicking-using-thiserror

I am not sure whether we should add the backtrace though - e.g. InvalidDep already tells you a lot of information, maybe they are already enough?

Reply:
Yeah, I've been thinking about what makes a backtrace useful in such cases. If the error is descriptive enough as to be self-diagnosable by the user then perhaps not. It is possible to add in your own backtraces though, so I can go back and add them in where needed.